substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/HeapPolicy.java

```diff
@@ -318,6 +318,11 @@ public static UnsignedWord getLargeArrayThreshold() {
         }
     }
 
+    @Fold
```

Comment:
This `@Fold` does not make much sense. It means that the option value of runtime option is only read and then constant folded during image generation.

You can either declare `MaxHeapFreeRatio` as a `HostedOptionKey` - then you can specify the option only during image generation and it is automatically constant folded even without the `@Fold` annotation.

Or you can remove the `@Fold` and get a "real" option that you can set at run time.

Reply:
Thanks, Christias! I removed the Fold. Remaining test failure doesn't look related.